astropy/units/tests/test_quantity_ufuncs.py

```diff
@@ -88,6 +92,30 @@ def test_removal_addition(self):
         assert np.add in qh.UFUNC_HELPERS
         assert np.add not in qh.UNSUPPORTED_UFUNCS
 
+    def test_thread_safety(self, fast_thread_switching):
+        def dummy_ufunc(*args, **kwargs):
+            return np.sqrt(*args, **kwargs)
+
+        def register():
+            return {dummy_ufunc: helper_sqrt}
+
+        workers = 8
+        with concurrent.futures.ThreadPoolExecutor(max_workers=workers) as executor:
+            for p in range(10000):
+                helpers = UfuncHelpers()
+                # UNSUPPORTED is (for some reason) a class attribute rather than an
```

Comment:
Hmm, what this solved by your lower-case version?  Sorry to only realize now, but it would seem to make sense to just initialize that in `__init__` as well.

Reply:
Yes, that's why I moved it from class attribute to instance attribute. I moved it back due to concern that it would break code that uses `UfuncHelpers.UNSUPPORTED`, although I've just searched Github and not found any hits for "UfuncHelpers" outside of astropy source code so it may be an unwarranted concern.

Looking at it again, is the `UfuncHelpers` name considered to be public API? It doesn't have an underscore, but it's also not listed in `__all__`.

Reply:
I'll defer to @mhvk here for a design decision. While you couldn't find usage on GitHub, doesn't mean people are not using it in their private code. But if we do decide to make break (pun?) for this, then we'll find out soon enough if people complain about it after the next release.

Reply:
My suggestion would be to leave it as upper case, but make it an instance property. Then, if anybody uses `UFUNC_HELPERS.UNSUPPORTED`, it will still work fine.

Reply:
p.s. So a bit in between - I don't think it is likely anybody would use the class, while I think it is quite possible they might have used the instance.

Reply:
Ok, I'll change it to an instance attribute tomorrow.

Reply:
Done.